src/enum_net.cpp

```diff
@@ -158,13 +158,20 @@ namespace libtorrent { namespace
 
 		do
 		{
-			int read_len = recv(sock, buf, bufsize - msg_len, 0);
+			int read_len = int(recv(sock, buf, bufsize - msg_len, 0));
 			if (read_len < 0) return -1;
 
 			nl_hdr = reinterpret_cast<nlmsghdr*>(buf);
 
+#if (TORRENT_ANDROID && defined __clang__)
+#pragma clang diagnostic push
```

Comment:
actually I now realize that this is a clang warning and it's correct since the struct types of `nlmsghdr` are unsigned. We don't see the warning in macOS because there is no `TORRENT_USE_NETLINK`.